model/buildlogger.go

```diff
@@ -464,87 +451,75 @@ func (l *Logs) Find(ctx context.Context, opts LogFindOptions) error {
 	} else {
 		return mongo.ErrNoDocuments
 	}
+	if opts.LatestExecution {
+		l.filterLatestExecution()
+	}
 
 	return nil
 }
 
+func (l *Logs) filterLatestExecution() {
```

Comment:
I don't follow the logic behind this. Is this trying to just get the single latest result?

If the results are already returned in order of decreasing execution ID, isn't it sufficient to just set `l.Logs` to the first element? And if they're not sorted by execution ID, don't you have to traverse the entire slice?

Reply:
So there can be more than one log document for an execution (actually there almost always is). For example, evergreen task logs, agent logs, and system logs that all have the same task id + execution but would be separate buildlogger metadata docs.